stdlib/src/mmap.rs

```diff
@@ -0,0 +1,1036 @@
+                _ => return Err(vm.new_value_error("unknown seek type".to_owned())),
+            };
+
+            if new_pos > size || new_pos < 0 {
+                return Err(vm.new_value_error("seek out of range".to_owned()));
+            }
+
+            self.pos.store(new_pos);
+
+            Ok(())
+        }
+
+        #[pymethod]
+        fn size(&self, vm: &VirtualMachine) -> PyResult<PyIntRef> {
+            let new_fd = unistd::dup(self.fd).map_err(|e| vm.new_os_error(e.to_string()))?;
+            let file = unsafe { File::from_raw_fd(new_fd) };
+            let file_len = match file.metadata() {
+                Ok(m) => m.len(),
+                Err(e) => return Err(vm.new_os_error(e.to_string())),
+            };
+
+            Ok(PyInt::from(file_len).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn tell(&self, vm: &VirtualMachine) -> PyResult<PyIntRef> {
+            Ok(PyInt::from(self.inner_pos()).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn write(&self, bytes: ArgBytesLike, vm: &VirtualMachine) -> PyResult<PyIntRef> {
+            let pos = self.inner_pos();
+            let size = self.inner_size();
+
+            let data = bytes.borrow_buf();
+
+            if pos > size || size - pos < data.len() as isize {
+                return Err(vm.new_value_error("data out of range".to_owned()));
+            }
+
+            let len = self.try_writable(vm, |mmap| {
+                (&mut mmap[pos as usize..(pos as usize + data.len())])
+                    .write(&data)
+                    .map_err(|e| vm.new_os_error(e.to_string()))?;
+                Ok(data.len())
+            })??;
+
+            self.advance_pos(len as isize);
+
+            Ok(PyInt::from(len).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn write_byte(&self, byte: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+            let b = value_from_object(vm, &byte)?;
+
+            let pos = self.inner_pos();
+            let size = self.inner_size();
+
+            if pos >= size {
+                return Err(vm.new_value_error("write byte out of range".to_owned()));
+            }
+
+            self.try_writable(vm, |mmap| {
+                mmap[pos as usize] = b;
+            })?;
+
+            self.advance_pos(1);
+
+            Ok(())
+        }
+
+        fn get_item_by_index(&self, i: isize, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+            let i = wrap_index(i, self.len())
+                .ok_or_else(|| vm.new_index_error("mmap index out of range".to_owned()))?;
+
+            let b = match self.check_valid(vm)?.deref().as_ref().unwrap() {
+                MmapObj::Read(mmap) => mmap[i as usize],
+                MmapObj::Write(mmap) => mmap[i as usize],
+            };
+
+            Ok(PyInt::from(b).into_ref(vm).into())
+        }
+
+        fn getitem_by_slice(
+            &self,
+            slice: &SaturatedSlice,
+            vm: &VirtualMachine,
+        ) -> PyResult<PyObjectRef> {
```

Comment:
```suggestion
        ) -> PyResult<PyBytes> {
```

Reply:
We can't change this return type to PyBytes, it have to be the same with `get_item_by_index`, both are used in `_getitem`.